text/287-cli-deprecation-warnings.md

```diff
@@ -0,0 +1,264 @@
+---
+rfc pr: [#290](https://github.com/aws/aws-cdk-rfcs/pull/290)
+tracking issue: https://github.com/aws/aws-cdk-rfcs/issues/287
+---
+
+# CDK CLI deprecation warnings
+
+If an element in the CDK Construct Library (class, interface, property, function) is deprecated,
+using it inside a CDK application should produce a warning in the CDK CLI for all commands that perform synthesis
+(`cdk synth`, `cdk diff`, `cdk deploy`, etc.).
+
+This is done in order to aid the migration from `V1` of the CDK to `V2`
+(where all deprecated elements from `V1` will be removed).
+
+## Working Backwards
+
+### CHANGELOG
+
+feat(cli): warn about usages of deprecated Construct Library elements
+
+### README
+
+When using any element (class, interface, property, function) that is deprecated in the Construct Library,
```

Comment:
Can we describe when this will _not_ work? For example, what does it mean to "use" an interface in this context?

Reply:
Since that depends on the exact solution chosen, I've omitted it from this part.

Do we need to get into such details in the user-facing documentation?

Reply:
This document should propose a solution and the "working backwards" section should be written as if this the proposed solution is the one we chose. If we end up changing the solution for whatever reason, we will need to evaluate the implications on the user experience and update this section (which is a good thing :-)).

Reply:
OK. Added. Let me know if this works.